File: app/channels/support_conversation_channel.rb
class SupportConversationChannel < ApplicationCable::Channel 
  # calls when a client connects to the server
  def subscribed
    puts "##############"
    puts params
    puts "##############"

    if params[:support_conversation_id].present?
      stream_from("support_conversation_#{(params[:support_conversation_id])}")
    else
      puts "support_conversation_id is missing."
    end
  end

  def unsubscribed
    stop_all_streams
  end
  
  # calls when a client broadcasts data
  def receive(data)
    sender = get_sender(data)
    support_conversation_id = data['support_conversation_id']
    message = data['body']

    raise 'No support_conversation_id!' if support_conversation_id.blank?
    convo = get_convo(support_conversation_id)
    raise 'No support conversation found!' if convo.blank?
    raise 'Body is blank!' if message.blank?
    convo.support_messages.create(body: message, sender_id: sender.id, user_id: current_user.id)
  end
  
  # Helpers
  
  def get_convo(support_conversation_id)
    SupportConversation.find_by(id: support_conversation_id)
  end
  
  def get_sender(data)
    User.find_by(id: data['sender_id'])
  end
  
end

File: app/javascript/channels/support_conversation_channel.js
import consumer from "./consumer"
$(document).ready(function() {
    const chat_id = $("#chatmessagebox").attr('data-id');
    const email = $("#admin_email").data("admin");
    console.log(chat_id);
    consumer.subscriptions.create({channel: "SupportConversationChannel", id: "support_conversation_" + chat_id}, {
        connected() {
            console.log("Connected SupportConversation Channel")
            // Called when the subscription is ready for use on the server
        },

        disconnected() {
            // Called when the subscription has been terminated by the server
        },

        received(data) {
            if(data.support_conversation_id== chat_id)
            {
                if(data.user_id == null)
                {
                    $(".msg_contain").append('<div class="msg admin">'+
                        '<div class="profile">'+
                        "<img src='/assets/default-profile.jpg'>"+
                        '</div>'+
                        '<div class="msg_wrapper">'+
                        '<div class="msg_blk">'+
                        '<p>'+data.body+'</p>'+
                        '</div>'+
                        '<p>'+data.created_at +'</p>'+
                        (data.message_image != null ?
                        '<div class="attch_blk">'+
                             "<img src="+data.message_image+">"+
                        '</div>'
                            : '')+
                        '</div>'

                    );


                }
                else
                {

                    $(".msg_contain").append('<div class="msg user">'+
                        '<div class="profile">'+
                        (data.sender_image == null ? "<img src="+data.sender_image+">" : "<img src='/assets/default-profile.jpg'>")+
                        '</div>'+
                        '<div class="msg_wrapper">'+
                        '<div class="msg_blk">'+
                        '<p>Open a Support Ticket <strong>'+
                        data.ticket_number.ticket_number+
                        '</strong></p>'+
                        '<p>'+data.body+'</p>'+
                        (data.message_image != null ?
                            '<div class="attch_blk">'+
                                "<img src="+data.message_image+">"+
                                '<div class="img">'+'<img src="assets/icon-download.svg" alt="">'+'</div>'+
                                    '<button type="button" class="btn down_btn">'+ '</button>'+
                                '</div>': '')+
                        '</div>'+'<p>'+data.created_at+'</p>'+
                        '</div>'+
                        '</div>');


                }
                $(".msg_contain").scrollTop($(".msg_contain")[0].scrollHeight);

            }
        }
    });

});
